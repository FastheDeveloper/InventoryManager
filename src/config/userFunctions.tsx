
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Alert} from 'react-native'

export const editByName = async (name: any, price: number, stock: number, desc: any, navigation: { navigate: (arg0: string) => void; }) => {
    try {
      const data = await AsyncStorage.getItem('items');
      if (data !== null) {
        const items = JSON.parse(data);
        const index = items.findIndex((item: { name: any; }) => item.name === name);
        if (isNaN(price) || isNaN(stock)) {
          Alert.alert('Invalid input', 'Price and stock must be numbers.');
          return;
        }
        if (index !== -1) {
          items[index] = { ...items[index], name, price, stock, desc };
          await AsyncStorage.setItem('items', JSON.stringify(items));
          Alert.alert('Success', 'Item updated successfully');
          navigation.navigate('Main');
        } else {
          Alert.alert('Not found', `Item with name ${name} not found`);
        }
      }
    } catch (error) {
      console.error(error);
    }
  };
  
  export const deleteByName = async (name: any, navigation: { navigate: (arg0: string) => void; }) => {
    try {
      const data = await AsyncStorage.getItem('items');
      if (data !== null) {
        const items = JSON.parse(data);
        const filteredItems = items.filter((item: { name: any; }) => item.name !== name);
        await AsyncStorage.setItem('items', JSON.stringify(filteredItems));
        Alert.alert('Success', 'Item deleted successfully');
        navigation.navigate('Main');
      } else {
        Alert.alert('Not found', `Item with name ${name} not found`);
      }
    } catch (error) {
      console.error(error);
    }
  };
  
  export const addItem = async (name: any, price: number, stock: number, desc: string, imageUrl: any, setName: (arg0: string) => void, setPrice: (arg0: number) => void, setStock: (arg0: number) => void, setDesc: (arg0: string) => void, navigation: { navigate: (arg0: string) => void; }) => {
    try {
      // Get the existing items from local storage
      const items = await AsyncStorage.getItem('items');
      let itemsArray = [];
      if (items !== null) {
        itemsArray = JSON.parse(items);
      }
  
      // Check if the item name is unique
      const existingItem = itemsArray.find((item: { name: any; }) => item.name === name);
      if (existingItem) {
        Alert.alert('Duplicate Item', 'An item with the same name already exists.');
        return;
      }
      if (isNaN(price) || isNaN(stock)) {
        Alert.alert('Invalid input', 'Price and stock must be numbers.');
        return;
      }
  
      const descriptionWords = desc.split(' ');
      if (descriptionWords.length < 3) {
        Alert.alert('Error', 'Description must have at least 3 words.');
        return;
      }
  
      // Add the new item to the items array
      const newItem = { name, price, stock, desc, imageUrl };
      itemsArray.push(newItem);
  
      // Save the updated items array to local storage
      await AsyncStorage.setItem('items', JSON.stringify(itemsArray));
  
      // Reset the input fields
      setName('');
      setPrice(0);
      setStock(0);
      setDesc('');
  
      Alert.alert('Success', 'The item has been added to the inventory.');
      navigation.navigate('Main');
    } catch (error) {
      
      Alert.alert('Error', 'Failed to add the item to the inventory.');
    }
  };
  