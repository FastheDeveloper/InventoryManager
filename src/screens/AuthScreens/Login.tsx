
import { Image, StyleSheet, Text, View ,Platform,Dimensions, TouchableOpacity, KeyboardAvoidingView} from 'react-native'
import React,{useContext, useEffect, useState} from 'react'
import { getStatusBarHeight } from 'react-native-status-bar-height';

import CustomInput from '../../components/inputs/AuthInput';
import MButtons from '../../components/MainButtons/Button';
import { AuthContext } from '../../config/AuthContext';

const statusBarHeight = getStatusBarHeight();
const { width, height } = Dimensions.get('window');
type Props = {
    navigation: any;
  };

const Login = ({ navigation }: Props) => {
  const [email, setEmail] = useState('');
  const [passin, setPassin] = useState('')
  const { login,authState } = useContext(AuthContext);

  const handleLogin = () => {
    login(email, passin);
    
  }
  console.log(authState.isLoggedIn,'Login');
  useEffect(() => {

    if (authState.isLoggedIn===true) {
      navigation.navigate('Main');
    }
  }, [authState.isLoggedIn, navigation]);
  

  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding">
    <View style={styles.innerContainer}>
    <View style={{ flexDirection:'row',alignItems:'center',justifyContent:'space-between'}}>
    <Text>{'                  '}</Text>
        <Text style={{fontSize:30,fontFamily:'PoppinsBold'}}>Login</Text>
        <Text>{'                  '}</Text>
    </View>

    <View style={{marginTop:20}}>
    
    <CustomInput label={'Email'} isEmail={true} onChangeText={(text: React.SetStateAction<string>) => setEmail(text)} isPassword={undefined} hidePassword={undefined} setHidePassword={undefined} isName={undefined}/>
    <CustomInput label={'Password'} isPassword={true} onChangeText={(text: React.SetStateAction<string>) => setPassin(text)} hidePassword={undefined} setHidePassword={undefined} isName={undefined} isEmail={undefined}/>
    </View>
    <View style={{alignItems:'center'}}>
    <TouchableOpacity onPress={handleLogin} >
    <MButtons label={'login'} disabled={undefined}/>
    </TouchableOpacity>
    </View>
   
      
    
    

    </View>
   
</KeyboardAvoidingView>
  )
}

export default Login

const styles = StyleSheet.create({
  container: { 
      flex: 1,   
      backgroundColor: '#fff',
    },
    innerContainer:{
      backgroundColor:'#B9DCA8',
      paddingTop: statusBarHeight+30,
      paddingHorizontal:15,
      height:height*0.86,
      borderBottomLeftRadius:200,
      borderBottomRightRadius:200,
      
    },
    signinView:{
      backgroundColor:'#fff',
      padding:5
    },
    icons:{
      flexDirection:'row',
      marginTop:41
     
      
    }
})