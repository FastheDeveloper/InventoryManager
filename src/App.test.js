import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Login from './screens/AuthScreens/Login';
import Edit from './screens/Main/Edit';
import AddItems from './screens/Main/AddItems';
import HomeScreen from './screens/Main/HomeScreen'
import InventoryScreen from './screens/Main/HomeScreen';
import { render, fireEvent } from '@testing-library/react-native';

import AsyncStorage from '@react-native-async-storage/async-storage';
import { addItem, editByName, deleteByName } from './config/userFunctions'

describe('Async storage CRUD operations', () => {
  beforeEach(async () => {
    await AsyncStorage.clear();
  });

  test('should add an item to async storage', async () => {
    const name = 'Test item';
    const price = 10;
    const stock = 5;
    const desc = 'This is a test item';
    const imageUrl = 'https://test.com/image.jpg';

    await addItem(name, price, stock, desc, imageUrl, jest.fn(), jest.fn(), jest.fn(), jest.fn());

    const data = await AsyncStorage.getItem('items');
    const items = JSON.parse(data);

    expect(items).toHaveLength(1);
    expect(items[0]).toEqual(expect.objectContaining({
      name: name,
      price: price,
      stock: stock,
      desc: desc,
      imageUrl: imageUrl
    }));
  });

  test('should update an item in async storage', async () => {
    const name = 'Test item';
    const price = 10;
    const stock = 5;
    const desc = 'This is a test item';
    const imageUrl = 'https://test.com/image.jpg';

    await addItem(name, price, stock, desc, imageUrl, jest.fn(), jest.fn(), jest.fn(), jest.fn());

    const newPrice = 20;
    const newStock = 10;
    const newDesc = 'This is an updated test item';

    await editByName(name, newPrice, newStock, newDesc, jest.fn());

    const data = await AsyncStorage.getItem('items');
    const items = JSON.parse(data);

    expect(items).toHaveLength(1);
    expect(items[0]).toEqual(expect.objectContaining({
      name: name,
      price: newPrice,
      stock: newStock,
      desc: newDesc,
      imageUrl: imageUrl
    }));
  });

  test('should delete an item from async storage', async () => {
    const name = 'Test item';
    const price = 10;
    const stock = 5;
    const desc = 'This is a test item';
    const imageUrl = 'https://test.com/image.jpg';

    await addItem(name, price, stock, desc, imageUrl, jest.fn(), jest.fn(), jest.fn(), jest.fn());

    await deleteByName(name, jest.fn());

    const data = await AsyncStorage.getItem('items');
    const items = JSON.parse(data);

    expect(items).toHaveLength(0);
  });

  test('should get an item from async storage', async () => {
    const name = 'Test item';
    const price = 10;
    const stock = 5;
    const desc = 'This is a test item';
    const imageUrl = 'https://test.com/image.jpg';

    await addItem(name, price, stock, desc, imageUrl, jest.fn(), jest.fn(), jest.fn(), jest.fn());

    const data = await AsyncStorage.getItem('items');
    const items = JSON.parse(data);
    const item = items.find((item) => item.name === name);

    expect(item).toEqual(expect.objectContaining({
      name: name,
      price: price,
      stock: stock,
      desc: desc,
      imageUrl: imageUrl
    }));
  });
});




//confirmNavigation
describe('InventoryScreen', () => {
    const testproduct = [{name: 'Item 1', price: 10, stock: 5, imageUrl: 'https://example.com/item1.jpg'},];
    const navigation = {
      navigate: jest.fn(),
    };
  
    afterEach(() => {
      jest.clearAllMocks();
    });
  
    it('Navigates successfully', () => {
      const { getByTestId } = render(
        <InventoryScreen testproduct={testproduct} navigation={navigation} />
      );
      const item1 = getByTestId('Item 1');
      fireEvent.press(item1);
      expect(navigation.navigate).toHaveBeenCalledWith('Edit', {
        itemName: 'Item 1',
      });
    });
  });
//confirmNavigation

//confirm popup
describe('Edit screen', () => {
  test('Confirmation Poppup Shown', () => {
    const navigation = {
      goBack: jest.fn(),
    };
    const route = {
      params: {
        itemName: 'Existing Item',
      },
    };

    const { getByText, getByTestId } = render(
      <Edit navigation={navigation} route={route} />
    );

    fireEvent.press(getByText('Delete by name'));

    const verifyModal = getByTestId('verify-modal');
    expect(verifyModal).toBeDefined();
  });
});
//confirm popup

//screenshots
describe('InventoryScreen', () => {
    
    it('Screenshot Inventory Home', () => {
      const tree = renderer.create(<InventoryScreen navigation={{ navigate: jest.fn() }} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    
  });


it('Screenshot Edit', () => {
    const tree = renderer.create(<Edit/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Screenshot Login', () => {
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Screenshot AddItems', () => {
    const tree = renderer.create(<AddItems/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
//screenshots

afterAll(() => {
    jest.useRealTimers();
  });
  