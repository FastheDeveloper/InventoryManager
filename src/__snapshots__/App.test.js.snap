// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryScreen Screenshot Inventory Home 1`] = `
<View>
  <View
    style={
      {
        "marginBottom": 150,
      }
    }
  >
    <RCTScrollView
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 50,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text>
          Add New item
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 50,
              },
              undefined,
              {
                "fontFamily": "evilicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text>
          Edit
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "red",
                "fontSize": 40,
              },
              undefined,
              {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text>
          Log out
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Screenshot AddItems 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "backgroundColor": "#B9DCA8",
        "borderBottomLeftRadius": 200,
        "borderBottomRightRadius": 200,
        "height": 1147.24,
        "paddingHorizontal": 15,
        "paddingTop": 50,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 28,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "fontFamily": "PoppinsBold",
            "fontSize": 30,
          }
        }
      >
        Add New Item
      </Text>
      <Text>
                          
      </Text>
    </View>
    <Text
      style={
        {
          "color": "red",
          "textAlign": "center",
        }
      }
    >
      Please fill all fields. they are required
    </Text>
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Name"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Price"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Total Stock"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Description"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 25,
                  "height": 48,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 60,
                },
                {
                  "backgroundColor": "rgba(69, 161, 24, 0.5)",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontFamily": "MontSBold",
                  "fontSize": 24,
                  "fontWeight": "700",
                }
              }
            >
              Add item
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 25,
                  "height": 48,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 60,
                },
                {
                  "backgroundColor": "pink",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontFamily": "MontSBold",
                  "fontSize": 24,
                  "fontWeight": "700",
                }
              }
            >
              Edit Inventory
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Screenshot Edit 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "backgroundColor": "#B9DCA8",
        "borderBottomLeftRadius": 200,
        "borderBottomRightRadius": 200,
        "height": 1147.24,
        "paddingHorizontal": 15,
        "paddingTop": 50,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 28,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "fontFamily": "PoppinsBold",
            "fontSize": 30,
          }
        }
      >
        Edit
      </Text>
      <Text>
                          
      </Text>
    </View>
    <Text
      style={
        {
          "color": "red",
          "textAlign": "center",
        }
      }
    >
      Please fill all fields. they are required
    </Text>
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Name"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Price"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Total Stock"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Description"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 25,
                  "height": 48,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 60,
                },
                {
                  "backgroundColor": "rgba(69, 161, 24, 0.5)",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontFamily": "MontSBold",
                  "fontSize": 24,
                  "fontWeight": "700",
                }
              }
            >
              Edit item by name
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 25,
                  "height": 48,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 60,
                },
                {
                  "backgroundColor": "pink",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontFamily": "MontSBold",
                  "fontSize": 24,
                  "fontWeight": "700",
                }
              }
            >
              Delete by name
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Screenshot Login 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "backgroundColor": "#B9DCA8",
        "borderBottomLeftRadius": 200,
        "borderBottomRightRadius": 200,
        "height": 1147.24,
        "paddingHorizontal": 15,
        "paddingTop": 50,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text>
                          
      </Text>
      <Text
        style={
          {
            "fontFamily": "PoppinsBold",
            "fontSize": 30,
          }
        }
      >
        Login
      </Text>
      <Text>
                          
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "black",
                    "fontSize": 24,
                  },
                  {
                    "marginLeft": 15,
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
            <TextInput
              onChangeText={[Function]}
              placeholder="Email"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 25,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(252, 252, 254, 0.5)",
              "borderColor": "#CED4DA",
              "borderRadius": 25,
              "borderWidth": 0.5,
              "justifyContent": "center",
              "paddingVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <TextInput
              onChangeText={[Function]}
              placeholder="Password"
              style={
                {
                  "fontSize": 17,
                  "marginLeft": 10,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 25,
                  "height": 48,
                  "justifyContent": "center",
                  "marginVertical": 10,
                  "paddingHorizontal": 60,
                },
                {
                  "backgroundColor": "rgba(69, 161, 24, 0.5)",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontFamily": "MontSBold",
                  "fontSize": 24,
                  "fontWeight": "700",
                }
              }
            >
              login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
